packages/mobile/src/screens/chat-screen/ChatScreen.tsx

```diff
@@ -231,15 +230,12 @@ export const ChatScreen = () => {
   const isLoading =
     (chat?.messagesStatus ?? Status.LOADING) === Status.LOADING &&
     chatMessages?.length === 0
-  // Only show the end reached indicator if there are no previous messages and
-  // flatlist is scrollable. Add an offset to the screen height because flatListInnerHeight
-  // starts at screen height due to height: 100% css property.
+  // Only show the end reached indicator if there are no previous messages, more than 10 messages,
+  // and the flatlist is scrollable.
   const shouldShowEndReachedIndicator =
     !(chat?.messagesSummary?.prev_count ?? true) &&
-    flatListInnerHeight.current - chatContainerTop.current >
-      chatContainerBottom.current -
-        chatContainerTop.current +
-        END_REACHED_OFFSET
+    chatMessages.length > END_REACHED_MIN_MESSAGES &&
```

Comment:
No longer need `END_REACHED_OFFSET` bc we don't use `minHeight: 100%` anymore.
Also removed `chatContainerTop` bc it's subtracted from both sides.

Reply:
Just curious why we need to do the math? Could we not just append an item to the end of the flatlist that renders the "end reached" message and let Flatlist handle the scrolling and positioning?

I've seen us use this pattern in tables where we give things special IDs (ex: `id: '__endReachedIndicator'`) that we can check for in `renderItem` to conditionally render something other than a ChatListMessageItem

Reply:
Good thought, I think we decided not to append things to flatlist for the unread indicator due to other issues but those might not apply here. Will create ticket.